src/include/sof/audio/codec_adapter/codec/generic.h

```diff
@@ -55,11 +55,19 @@ struct codec_config {
 	void *data; /* tlv config */
 };
 
+struct codec_memory {
+	void *ptr;
+	struct codec_memory *prev;
+	struct codec_memory *next;
+};
+
```

Comment:
should we use list API here? (see list.h?)

Reply:
Eh, in this particular case a bespoke list might be acceptable. I'd say leave it up to him.

Reply:
@paulstelian97 Ok, no strong preference on this.

Reply:
We can use the `list.h` API.